src/Symfony/Component/HttpClient/Psr18Client.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpClient;
+
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Client\NetworkExceptionInterface;
+use Psr\Http\Client\RequestExceptionInterface;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseFactoryInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\StreamFactoryInterface;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * An adapter to turn a Symfony HttpClientInterface into a PSR-18 ClientInterface.
+ *
+ * Run "composer require psr/http-client" to install the base ClientInterface. Run
+ * "composer require nyholm/psr7" to install an efficient implementation of response
+ * and stream factories with flex-provided autowiring aliases.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+final class Psr18Client implements ClientInterface
```

Comment:
I can move it to the bridge. Please advise. I don't really care personally.

Reply:
I would go for consistency. Either all adapters inside the components they're wrapping or everything into a bridge package.

The more I think about it the more I'd like to keep `Psr18Client` in the main repo because I would want the framework to provide PSR-18 wrapped http clients.

The bridge on the other hand could need some love (it currently implements PSR-7, but should be updated for PSR-15 and PSR-17). Maybe we could create a bridge package for all HTTP-related PSR adapters inside the main repo?

Reply:
I would prefer, keeping the bridge inside this component, or if in another component keeping it alone. It would be weird for an user to have a lot of bridges and other abstractions where he only want an http client.

(it's like mixing http client and http foundation into the same component, no sense at all imo).

Also don't think there will be ever be a need for a PSR17 bridge since it's only for library implementing or using PSR7 which will never be the case for Symfony.

Reply:
On 2nd though, I agree, the class is fine here: nice fit with the domain, easier dependency management, easier integration with autowiring.